File: backend/models/PlanUser.js
const mongoose = require('mongoose');

const planUserSchema = new mongoose.Schema({
  planId: { type: String, ref: 'Plan', required: true },  // Ref to Plan model
  userId: { type: String, ref: 'User', required: true },
  role: { type: String, enum: ['VibeCoordinator', 'VibePlanner', 'Wanderer'], required: true }
}, { timestamps: true });

// Unique index for multiple roles per plan
planUserSchema.index({ planId: 1, userId: 1 }, { unique: true });

module.exports = mongoose.model('PlanUser', planUserSchema);

File: frontend/lib/models/plan_user.dart
import '../models/user.dart'; // For UserRole enum

class PlanUser {
  final String planId;
  final String userId;
  final UserRole role;

  PlanUser({required this.planId, required this.userId, required this.role});

  factory PlanUser.fromJson(Map<String, dynamic> json) {
    return PlanUser(
      planId: json['planId'] ?? '',
      userId: json['userId'] ?? '',
      role: _roleFromString(json['role'] ?? 'wanderer'), // Parse string to enum
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'planId': planId,
      'userId': userId,
      'role': role.toString().split('.').last, // Enum to string for backend
    };
  }

  // Helper to parse string to UserRole
  static UserRole _roleFromString(String roleStr) {
    switch (roleStr.toLowerCase()) {
      case 'vibecoordinator':
        return UserRole.vibeCoordinator;
      case 'vibeplanner':
        return UserRole.vibePlanner;
      default:
        return UserRole.wanderer;
    }
  }
}
